### PR TITLE
Fixes an issue when navigating to Stats on iPad from onboarding questions

### LINKED ISSUE #18985: [iOS] Stats screen shown incorrectly on iPad
### Expected behavior

The Stats screen should always be shown on the right side of the iPad (secondary pane).

### Actual behavior

The Stats screen is shown on the left side of the iPad (primary pane), replacing the My Site Home/Menu. This happens when users log in and choose the "focus" called checking stats.

### Steps to reproduce the behavior

1. Ensure you are logged out of the app
2. Log into the app
3. After choosing a site, the app will ask you what focus you would like to pick
4. Choose "checking stats" (or a similarly named item)
5. Optionally enable notifications
6. Notice that the Stats screen is shown on the left and right sides of the screen

### Screenshot

Screen Shot 2022-07-04 at 15 16 26

### Additional information

I’m not 100% sure this bug is related to the recent Stats Revamp project. @frosty found that it was reproducible even when the Stats Revamp feature flag was off.

### PR DESCRIPTION
**Pull request by frosty**
Fixes #18985. This PR updates the onboarding prompts Stats navigation to use the newer navigation method that has been fixed to handle the home screen Dashboard.

**To test**

* Build and run on iPad (also verify on iPhone afterwards)
* Either fresh install, or log out if you are already logged in
* Log in, and on the final onboarding step choose "Checking Stats"
* You should be taken to the My Sites screen, with the menu shown in the left pane and Stats shown in the right
* On iPhone, you should just be taken to Stats, and be able to navigate back up the stack to My Sites.

## Regression Notes
1. Potential unintended areas of impact

None

2. What I did to test those areas of impact (or what existing automated tests I relied on)
3. What automated tests I added (or what prevented me from doing so)

N/A

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by guarani** [comment]
@frosty, due to our timezone differences and the upcoming new beta for 20.2, I'd like to merge this after reviewing.

**Review by guarani — COMMENTED** [review]
I tested this and ran into an issue where the Stats Revamp Feature Announcement only shows momentarily before being unexpectedly closed by the app.  

1. Log in
2. Choose Stats as your focus
3. Notice the Stats Revamp Feature Announcement appears briefly before being dismissed

**Comment by mokagio** [comment]
@frosty @guarani I changed the base branch to `trunk` and moved this to the 20.3 milestone because, as discussed the stats changes will no longer be part of 20.2. See #18994 .

Please revert to the original settings if I made a bundler, which is something I now wonder give this is a navigation change, not a UI one 🤔

**Comment by guarani** [comment]
You're totally right to update the milestone, thanks @mokagio!

**Comment by mokagio** [comment]
Pushing this PR to the next milestone, 20.4, because I'm running the 20.3 code-freeze today and this hasn't been approved yet.

However, I see this is a bugfix PR. I'll leave it up to you to judge whether the path it touches is high-occurrence enough to grant the extra work needed to land it on 20.3, or if it can wait till 20.4. That is to say, feel free to change the milestone back to 20.3 and I'll ship this fix as part of a new beta when it lands. 👍

**Comment by sla8c** [comment]
The fix looks good for me (with and without stats revamp feature flag) ie its unrelated to stats revamp work.

I tested:
 * on iPhone device with and without stats revamp feature flag
 * on iPad simulator with and without stats revamp feature flag
 * and compared to both AppStore version and TestFlight v20.3

**Review by sla8c — APPROVED** [review]

**Pull request merged**
